storage/series_cursor.go

```diff
@@ -98,6 +103,9 @@ func (cur *seriesCursor) Next() (*SeriesCursorRow, error) {
 		}
 
 		cur.row.Name, cur.row.Tags = tsdb.ParseSeriesKey(cur.keys[cur.ofs])
+		if !bytes.HasPrefix(cur.row.Name, cur.orgID) {
```

Comment:
I think we should either change the `orgID` type to something stronger, or if we leave `orgID` as a `[]byte` then maybe `len(cur.row.Name) < 8 || !bytes.Equal(cur.row.Name[:8], cur.orgID)`.

Reply:
`bytes.HasPrefix` is implemented as 

```go
len(name) >= len(orgID) && Equal(name[0:len(prefix)], orgID)
```

So by adding an assertion in `newSeriesCursor`, such as `len(orgID) == influxdb.IDBytesLen`, we could ensure this invariant.